### PR TITLE
Prog view

### PR DESCRIPTION
**Pull request by ryanmadamba**
Student Progress Overview Page.

I tried my best to add comments and specify where code for Mastery Progress (Learning Targets) and Mission Progress.

### CODE REVIEW

File: src/Components/ProgressOverview.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+//Query Mission Data
+const { loading, error, data, refetch} = useQuery(GET_MISSIONS, {
+  variables: { id: "Integrated Science" }
+});
+
+//Query Mission Progress Data (not currently user - using hard coded data above)
+const { loading: progressLoading, error: progressError, data: progressData, refetch : progressRefetch} = useQuery(GET_ALL_MISSION_PROGRESS, {
+  variables: { id: " " }
+});
```
Comment (ConnorBoulais) [inline src/Components/ProgressOverview.js:63]: Nice job hooking up to the backend here.

File: src/Components/ProgressOverview.js [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  }
+  //progress object not found
+  if (progress == null){
+    console.warn("Mission progress not found.");
+    return(
+      <div className="card">
+        <h1>No data found!</h1>
+      </div>
+    );
+  }
```
Comment (ConnorBoulais) [inline src/Components/ProgressOverview.js:117]: nice logic here

File: src/Components/Courses.js [hunk 3]
```diff
@@ -77,16 +72,6 @@ function Course() {
           </div>
         )
       } 
-      /*else {
```
Comment (tddumont) [inline src/Components/Courses.js:80]: Thanks for cleaning this up!

File: src/Components/ProgressOverview.js [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+//Used to select which Mission Progress information should be shared
+function changeFocusedMission(mission){
+  console.log(focusedMission);
+  setFocusedMission({
+    id: mission.id,
+    name: mission.name,
+    description: mission.description
+  })
+}
```
Comment (ConnorBoulais) [inline src/Components/ProgressOverview.js:155]: Good with the focused mission too

**Review by ConnorBoulais — APPROVED** [review]
Great job here, this is a solid start on the progress screen. Nice hooking it up to the backend as well.

File: src/Components/ProgressOverview.js [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+      <div key={task.id}>
+        <ul>
+          {task.name}
+        </ul>
+      </div>
+    )        
+  });
+}
+
+//----------- Mastery Progress (Learning Targets)----------
```
Comment (tddumont) [inline src/Components/ProgressOverview.js:182]: Thanks for leaving a comment to divide this up!

File: src/Components/ProgressOverview.js [hunk 6]
```diff
@@ -0,0 +1,285 @@
+import { useQuery} from '@apollo/client';
+import React, { useState } from 'react';
+import ProgressBar from './ProgressBar.js'
+import './ProgressOverview.css'
+import { GET_ALL_MISSION_PROGRESS, GET_MISSIONS, GET_ALL_TARGET_PROGRESS, GET_TARGETS } from '../gqlQueries.js';
+
+//This component is used to display the Student Progress Overview.
+function ProgressOverview() {
```
Comment (tddumont) [inline src/Components/ProgressOverview.js:8]: This file has a lot of lines, but this will be broken up in the future.

**Review by tddumont — APPROVED** [review]
Looks great!

**Pull request merged**